source/common/common/status.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
```

Comment:
What would be the split between common and codec specific? We can do different things here:

1.  Move everything to the `common/http` for now as there is currently no use-case outside of codecs.
1.  Leave `Status` alias and `StatusCode` in `common/common` and move all code specific error making and checking functions into `common/http`.
1.  Introduce component specific `StatusCode` that maps to different `absl::StatusCode`. This could be fragile. I would really prefer single error space for all components.

Reply:
I would probably vote for (1) or (2) but I don't feel strongly about it if you prefer to keep it like this.

Reply:
Moved to `common/http`